### PR TITLE
Setup Sentry

### PR DESCRIPTION
**Pull request by denys-medynskyi**
`sentrygin` was used to introduce gin-specific middleware to set tra. More [can be read here](https://docs.sentry.io/platforms/go/guides/gin/#usage).

### How to test
- Add `SENTRY_DSN` to `.env`
- Introduce panic, for example open `controllers/file_controller.go`
```
func a(span *sentry.Span) {
	span.Status = sentry.SpanStatusFailedPrecondition
	err := errors.New("Sample Error")
	sentry.CaptureException(err)

	panic("error in a")
}

func (f *fileController) ParseXLSXFile(ctx *gin.Context) {
  ... sentry code
  a()
}
```
- start server `go run main.go`
- trigger panic with CURL
```
curl -X POST http://localhost:4000/api/uploadXlsx   -F "files=dsp-tax-saver-fund.xlsx"
```
- See in log Sentry sents events
- Screenshot 2024-10-04 at 5 39 18 PM
- Go to Sentry
- Verify details for panic are there
- Screenshot 2024-10-08 at 4 11 08 PM

- **Result:** issue should be visible with all the context

### CODE REVIEW

**Comment by shivamsouravjha** [comment]
There’s been a major code refactor.

**Comment by denys-medynskyi** [comment]
@shivamsouravjha I see, rebased

File: main.go [hunk 1]
```diff
@@ -78,6 +82,26 @@ func GracefulShutdown(server *http.Server, ticker *time.Ticker) {
...
+		EnableTracing: true,
+		Debug:         true,
+		// Set TracesSampleRate to 1.0 to capture 100%
+		// of transactions for tracing.
+		// Sentry recommend adjusting this value in production,
+		TracesSampleRate: tracesSampleRate, // 1.0 by default if ENV SENTRY_SAMPLE_RATE not set
+	}); err != nil {
+		fmt.Printf("Sentry initialization failed: %v\n", err)
+	}
+}
```
Comment (shivamsouravjha) [inline main.go:103]: lets use logger instead of print

**Comment by shivamsouravjha** [comment]
if we pass the sentry context across different level we will get more indepth analysis.

If we could pass the context we get this kind of analysis https://github.com/shivamsouravjha/PiePay (check the readme)

File: controllers/file_controller.go [hunk 2]
```diff
@@ -18,9 +20,20 @@ type fileController struct{}
...
+		transaction.Name = "ParseXLSXFile"
+	}
+
+	span := sentry.StartSpan(context.TODO(), "ParseXLSXFile")
+	defer span.Finish()
+
 	// Parse the form and retrieve the uploaded files
 	form, err := ctx.MultipartForm()
 	if err != nil {
+		span.Status = sentry.SpanStatusFailedPrecondition
```
Comment (denys-medynskyi) [inline controllers/file_controller.go:35]: @shivamsouravjha added, please check

**Comment by shivamsouravjha** [comment]
Hey @denys-medynskyi thanks for the update, the file_services still don't have the sentry because of which sentry logs would be around controlllers only.

**Comment by denys-medynskyi** [comment]
@shivamsouravjha added panic, screenshots are updated. It works.

**Pull request merged**